src/test/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractServiceImplTest.java

```diff
@@ -0,0 +1,118 @@
+
+import java.sql.Date;
+import java.util.Calendar;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.kuali.kfs.krad.exception.ValidationException;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+import edu.cornell.kfs.concur.ConcurConstants;
+import edu.cornell.kfs.concur.batch.businessobject.ConcurStandardAccountingExtractDetailLine;
+import edu.cornell.kfs.concur.batch.businessobject.ConcurStandardAccountingExtractFile;
+
+public class ConcurStandardAccountingExtractServiceImplTest {
+    
+    ConcurStandardAccountingExtractServiceImpl concurStandardAccountingExtractService;
+
+    @Before
+    public void setUp() throws Exception {
+        Logger.getLogger(ConcurStandardAccountingExtractServiceImpl.class).setLevel(Level.DEBUG);
+        concurStandardAccountingExtractService = new ConcurStandardAccountingExtractServiceImpl();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        concurStandardAccountingExtractService = null;
+    }
+    
+    @Test
+    public void validateDetailCountGood() {
+        ConcurStandardAccountingExtractFile file = buildConcurStandardAccountingExtractFile();
+        try {
+            concurStandardAccountingExtractService.validateDetailCount(file);
+            assertTrue("We successfully validated counts.", true);
+        } catch (ValidationException ve) {
+            assertTrue("The counts should be been the same.", false);
+        }
+    }
+    
+    @Test
+    public void validateDetailCountIncorrectMatch() {
+        ConcurStandardAccountingExtractFile file = buildConcurStandardAccountingExtractFile();
+        file.setRecordCount(new Integer(5));
+        try {
+            concurStandardAccountingExtractService.validateDetailCount(file);
+            assertTrue("The counts should not be equal", false);
+        } catch (ValidationException ve) {
+            assertTrue("We successfully validated counts.", true);
+        }
+    }
+    
+    @Test
+    public void validateAmountsGood() {
+        ConcurStandardAccountingExtractFile file = buildConcurStandardAccountingExtractFile();
+        String message = "The amounts should equal";
+        try {
+            concurStandardAccountingExtractService.validateAmounts(file);
+            assertTrue(message, true);
+        } catch (ValidationException ve) {
+            assertTrue(message, false);
+        }
+    }
+    
+    @Test
+    public void validateAmountsAmountMismatch() {
+        ConcurStandardAccountingExtractFile file = buildConcurStandardAccountingExtractFile();
+        file.getConcurStandardAccountingExtractDetailLines().get(0).setJournalAmount(new KualiDecimal(200));
+        String message = "The amounts should NOT equal and throw a validation error";
+        try {
+            concurStandardAccountingExtractService.validateAmounts(file);
+            assertTrue(message, false);
```

Comment:
sorry I missed this earlier, but the way the tests are written now they always pass. I think you will want to split these in 2 tests, one for success and one for failure.